include/cetl/any.hpp

```diff
@@ -49,41 +53,97 @@ struct base_storage  // NOLINT(*-pro-type-member-init)
     template <typename Tp>
     void make_handlers() noexcept
     {
-        assert(nullptr == value_destroyer_);
+        static_assert(sizeof(Tp) <= Footprint, "Enlarge the footprint");
+
+        CETL_DEBUG_ASSERT(nullptr == value_destroyer_, "Expected to be empty before making handlers.");
+        CETL_DEBUG_ASSERT(nullptr == value_converter_, "");
+        CETL_DEBUG_ASSERT(nullptr == value_const_converter_, "");
+        CETL_DEBUG_ASSERT(nullptr == value_type_informer_, "");
 
         value_destroyer_ = [](void* const storage) {
             const auto ptr = static_cast<Tp*>(storage);
             ptr->~Tp();
         };
+
+        make_converters<Tp>();
+
+        value_type_informer_ = [](const void*) {
+
+#if __cpp_rtti
+            return static_cast<const void*>(&typeid(Tp));
+#else
+            return static_cast<const void*>(nullptr);
+#endif
+        };
+    }
+
+    template <typename Tp, std::enable_if_t<is_rtti_convertible<Tp>, int> = 0>
+    void make_converters() noexcept
+    {
+        value_const_converter_ = [](const void* const storage, const type_id& id) {
+            const auto ptr = static_cast<const Tp*>(storage);
+            return ptr->_cast_(id);
+        };
+        value_converter_ = [](void* const storage, const type_id& id) {
+            auto ptr = static_cast<Tp*>(storage);
+            return ptr->_cast_(id);
+        };
+    }
+
+    template <typename Tp, std::enable_if_t<!is_rtti_convertible<Tp>, int> = 0>
+    void make_converters() noexcept
+    {
+        value_const_converter_ = [](const void* const storage, const type_id& id) {
+            return (id == type_id_value<Tp>) ? storage : nullptr;
+        };
+        value_converter_ = [](void* const storage, const type_id& id) {
+            return (id == type_id_value<Tp>) ? storage : nullptr;
+        };
+    }
+
+#if __cpp_rtti
+    CETL_NODISCARD const std::type_info* get_type_info() const noexcept
+    {
+        CETL_DEBUG_ASSERT(nullptr != value_type_informer_, "");
+
+        return static_cast<const std::type_info*>(value_type_informer_(get_raw_storage()));
     }
+#endif
 
     template <typename ValueType>
     CETL_NODISCARD void* get_ptr() noexcept
     {
+        static_assert(sizeof(ValueType) <= Footprint, "Enlarge the footprint");
```

Comment:
I think it would be more accurate to say that the type cannot contain the requested type by its definition since the footprint would be too small. The suggestion to enlarge the footprint is only meaningful in a context where one is attempting to place a large object into the `any` container.

Same comment applies below.

Reply:
ok, you are right, I'll change them to "Cannot contain the requested type since the footprint is too small".